#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_debian.dpatch by Petter Reinholdtsen <pere@hungry.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adjust some paths to get this working on debian.

@DPATCH@
--- insserv-1.00.8.orig/insserv.c
+++ insserv-1.00.8/insserv.c
@@ -1023,15 +1088,15 @@
 	struct stat st_script;
 
 	switch (runlevel) {
-	    case 0: rcd = "rc0.d/";  break;
-	    case 1: rcd = "rc1.d/";  break;
-	    case 2: rcd = "rc2.d/";  break;
-	    case 3: rcd = "rc3.d/";  break;
-	    case 4: rcd = "rc4.d/";  break;
-	    case 5: rcd = "rc5.d/";  break;
-	    case 6: rcd = "rc6.d/";  break;
-	    case 7: rcd = "rcS.d/";  break;  /* runlevel S */
-	    case 8: rcd = "boot.d/"; break;  /* runlevel B */
+	    case 0: rcd = "../rc0.d/";  break;
+	    case 1: rcd = "../rc1.d/";  break;
+	    case 2: rcd = "../rc2.d/";  break;
+	    case 3: rcd = "../rc3.d/";  break;
+	    case 4: rcd = "../rc4.d/";  break;
+	    case 5: rcd = "../rc5.d/";  break;
+	    case 6: rcd = "../rc6.d/";  break;
+	    case 7: rcd = "../rcS.d/";  break;  /* runlevel S */
+	    case 8: rcd = "../rc.boot/"; break;  /* runlevel B */
 	    default:
 		error("Wrong runlevel %d\n", runlevel);
 	}
@@ -1846,15 +1911,15 @@
 	DIR  * rcdir;
 
 	switch (runlevel) {
-	    case 0: rcd = "rc0.d/";  break;
-	    case 1: rcd = "rc1.d/";  break;
-	    case 2: rcd = "rc2.d/";  break;
-	    case 3: rcd = "rc3.d/";  break;
-	    case 4: rcd = "rc4.d/";  break;
-	    case 5: rcd = "rc5.d/";  break;
-	    case 6: rcd = "rc6.d/";  break;
-	    case 7: rcd = "rcS.d/";  break;  /* runlevel S */
-	    case 8: rcd = "boot.d/"; break;  /* runlevel B */
+	    case 0: rcd = "../rc0.d/";  break;
+	    case 1: rcd = "../rc1.d/";  break;
+	    case 2: rcd = "../rc2.d/";  break;
+	    case 3: rcd = "../rc3.d/";  break;
+	    case 4: rcd = "../rc4.d/";  break;
+	    case 5: rcd = "../rc5.d/";  break;
+	    case 6: rcd = "../rc6.d/";  break;
+	    case 7: rcd = "../rcS.d/";  break;  /* runlevel S */
+	    case 8: rcd = "../rc.boot/"; break;  /* runlevel B */
 	    default:
 		error("Wrong runlevel %d\n", runlevel);
 	}
@@ -1904,7 +1969,7 @@
 	    if (*script == '$')		/* Do not link in virtual dependencies */
 		continue;
 
-	    sprintf(olink, "../%s",   script);
+	    sprintf(olink, "../init.d/%s",   script);
 	    sprintf(nlink, "S%.2d%s", order, script);
 
 	    found = false;
